deserializer.go

```diff
@@ -4,9 +4,12 @@ import (
 	"bytes"
 	"encoding/gob"
 	"regexp"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/getlantern/golog"
+	iradix "github.com/hashicorp/go-immutable-radix"
```

Comment:
I was just asking because immutable implementations often have a bit of a performance penalty. IIRC the main use of "immutable" data structures is actually for state that you want to be able to mutate but allow old references to the state to continue to be able to use the old version without locking. In this particular case, we're not actually interested in that sort of benefit since, as you say, we don't change anything post-startup. Would be interesting to do a quick microbenchmark to see if there's a significant memory or cpu penalty vs https://github.com/armon/go-radix.